apolloschurchapp/src/ui/Onboarding/index.js

```diff
@@ -57,43 +59,48 @@ function Onboarding({ navigation }) {
             <LocationFinderConnected
               onPressPrimary={swipeForward}
               onNavigate={() => {
-                navigation.navigate('Location');
+                navigation.navigate('Location', { onFinished: swipeForward });
               }}
               BackgroundComponent={
                 <StyledGradient
                   source={'https://picsum.photos/640/640/?random'}
                 />
               }
             />
-            <AskNotificationsConnected
-              onRequestPushPermissions={(update) => {
-                checkNotifications().then((checkRes) => {
-                  if (checkRes.status === RESULTS.DENIED) {
-                    requestNotifications(['alert', 'badge', 'sound']).then(
-                      () => {
-                        update();
+            <Query query={WITH_USER_ID} fetchPolicy="network-only">
+              {({
+                data: { currentUser: { id } = { currentUser: { id: null } } },
+              }) => (
+                <AskNotificationsConnected
+                  description={
+                    'Get updates, reminders, and announcements from your Church.'
+                  }
+                  onPressPrimary={() => {
+                    onboardingComplete({ userId: id });
+                    navigation.replace('Tabs');
```

Comment:
We should keep the `dispatch(NS.resetAction)` that was here before. Willow should be updated to use the `dispatch(NS.resetAction)` eventually :) `.replace()` lets you hit back and return to the onboarding screens after logging in)